### PR TITLE
End of weekend - Twilio code working but hidden from github

### PR DESCRIPTION
**Pull request by DavidStewartLDN**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,4 +5,5 @@ group :test do
   gem 'rubocop', '0.72.0'
   gem 'simplecov', require: false, group: :test
   gem 'simplecov-console', require: false, group: :test
+  gem 'twilio-ruby'
```
Comment (EWright212) [inline Gemfile]: congrats on learning to edit gemfiles!

File: README.md [hunk 2]
```diff
@@ -14,21 +14,53 @@ Takeaway Challenge
...
-* You must submit a pull request to this repo with your code by 9am Monday morning
+I created this application for second weekend challenge at [Makers Academy](https://github.com/makersacademy).
 
-Task
------
 
-* Fork this repo
-* Run the command 'bundle' in the project directory to ensure you have all the gems
-* Write a Takeaway program with the following user stories:
+## Instructions
```
Comment (EWright212) [inline README.md]: Great job on adding user instructions to your README, useful to have :)

File: ideas.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+class Takeway
```
Comment (EWright212) [inline ideas.rb]: Could possibly delete this file now that you have the code in other files

File: lib/calculator.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+class Calculator
+
+  attr_reader :result, :total
+
+  # Would like to shorten method but not sure how
+  def self.current_basket(order, dishes)
```
Comment (EWright212) [inline lib/calculator.rb]: I love your ability to check what's in your basket

File: lib/menu.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class Menu
```
Comment (EWright212) [inline lib/menu.rb]: Separating the Menu into a separate class makes it super readable

File: lib/takeaway.rb [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  
+  def initialize(menu: Menu.new, sms_texter: SmsTexter.new)
+    @menu = menu
+    @order = {}
+    @sms_texter = sms_texter
+  end
+
+  def add(item, quantity = 1)
+    item = item.to_sym
+    raise "Item not in menu, please check spelling" unless @menu.dishes.key?(item)
```
Comment (EWright212) [inline lib/takeaway.rb]: Good edge case

File: lib/takeaway.rb [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    iterate_through_order(item, quantity)
```
Comment (EWright212) [inline lib/takeaway.rb]: I didn't know this was a thing but it's great

File: lib/takeaway.rb [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+  def basket(dishes = @menu.dishes)
+    input = Calculator.current_basket(@order, dishes)
+    output = []
+    input.each do |item|
+      output.push("#{item.values[0]} x#{item.values[1]} = £#{item.values[2]}")
+    end
+    output.join(", ")
+  end
+
+  def total
```
Comment (EWright212) [inline lib/takeaway.rb]: This is great for the user to be able to check the total without calculating it

File: spec/calculate_spec.rb [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      expect(Calculator).to respond_to(:current_basket).with(2).argument
+    end
+
+    it "should return a hash with the item_name, quantity and subtotal" do
+      order = {"Pilau rice": 1}
+      dishes = {"Pilau rice": 3.00}
+      expect(Calculator.current_basket(order, dishes)).to eq ([{item_name: "Pilau rice", quantity: 1, subtotal: 3.0}])
+    end
+
+    it "should return a hash with the item_name, quantity and subtotal for MULTIPLE items" do
```
Comment (EWright212) [inline spec/calculate_spec.rb]: Maybe you could add a test for three dishes ordered? Or a test for an edge case where no dishes ordered

File: spec/sms_texter_spec.rb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+require "sms_texter"
```
Comment (EWright212) [inline spec/sms_texter_spec.rb]: Good job having a separate class for the SMS tester and hiding it from GitHub

File: README.md [hunk 11]
```diff
@@ -14,21 +14,53 @@ Takeaway Challenge
...
+$ irb
+> require './lib/takeaway.rb'
+```
+
+#### Create an takeaway ####
+
+You can create a takeaway using the following command and with a result like this. Yes we have the paneer here, have no fear:
+
+```Shell
+> t = Takeaway.new # => <Takeaway:0x00007fbceb41cd78 @menu=#<Menu:0x00007fbceb41cd28 @dishes={:"Kolhapuri Murgi"=>5.6, :"Salmon Tikka Masala"=>8.8, :"Butter Chicken"=>6.1, :"Saag Paneer"=>3.0, :"Pilau rice"=>1.5, :"Coconut rice"=>1.8, :Poppadom=>0.5, :"Mango Chutney"=>0.3}>, @order={}, @sms_texter=#<SmsTexter:0x00007fbceb41ccb0>>
```
Comment (EWright212) [inline README.md]: Nice job for using symbols

File: README.md [hunk 12]
```diff
@@ -14,21 +14,53 @@ Takeaway Challenge
...
+
+#### See something you like? add it to your basket ####
+
+If you see something on the menu that takes your fancy, swipe it up quick with the add function. You by default you will add one to your basket but if you're feeling peckish then you can order multiple at a time by adding a quantity after the item you want to add.
+
+```Shell
+> t.add "Kolhapuri Murgi" # => {:"Kolhapuri Murgi"=>1}
+> t.add "Coconut rice", 10 # => {:"Kolhapuri Murgi"=>1, :"Coconut rice"=>10} 
+```
+That's a lot of rice for one person... :|
```
Comment (EWright212) [inline README.md]: Hahaha

File: ideas.rb [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  def initialize(menu = Menu.new, basket = Basket.new, text_providor = TextProvidor.new)
```
Comment (EWright212) [inline ideas.rb]: Maybe you could find another way to link your classes, as I think best practice is that each class works and can be tested separately without relying on each other

File: lib/takeaway.rb [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+  def iterate_through_order(item, quantity)
+    @order.key?(item) ? iterate_and_update(item, quantity) : @order[item] = quantity
+    @order
+  end
+
+  def iterate_and_update(item, quantity)
+    @order.each do |key, value|
+      @order[item] = value + quantity if item == key
+    end
+  end
```
Comment (EWright212) [inline lib/takeaway.rb]: these two methods are slightly hard to understand exactly what they do - could you add a comment for each one?
